gloo/test/CMakeLists.txt

```diff
@@ -28,7 +30,7 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
 endif()
 
 add_executable(gloo_test ${GLOO_TEST_SRCS})
-target_link_libraries(gloo_test gloo gtest ${GLOO_TEST_LIBRARIES})
+target_link_libraries(gloo_test gloo gtest ${GLOO_TEST_LIBRARIES} OpenSSL::SSL)
```

Comment:
why do we need it?

Reply:
The openssl_utils needs this or else it won't compile

Reply:
It would not compile or would not link? If any of those, @pbelevich can you please take care of fixing the tests?